go/codegen/pai/tensorflow.go

```diff
@@ -236,7 +236,20 @@ func getCheckpointDir(ossModelPath, project string) (string, error) {
 		return "", err
 	}
 	ossURL := OSSModelURL(ossModelPath)
-	roleName := fmt.Sprintf("pai2oss_%s", project)
+	roleName := genRoleName(project)
 	// format the oss checkpoint path with ARN authorization.
 	return fmt.Sprintf("%s/?role_arn=%s/%s&host=%s", ossURL, ra.Arn, roleName, ra.Host), nil
 }
+
+func genRoleName(project string) string {
+	var rn bytes.Buffer
+	rn.WriteString("pai2oss")
+	for _, ch := range project {
```

Comment:
How about using `regexp` to replace this loop like:
```Go
re := regexp.MustCompile(`[^a-z0-9]`)
rn.WriteString(re.ReplaceAllString(strings.ToLower(project), ""))
```

Reply:
It is a concise way, especially one pass, and easy to understand.
So I didn't choose using `regexp`.